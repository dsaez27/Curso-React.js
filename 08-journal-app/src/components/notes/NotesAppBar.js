import React from 'react'

export const NotesAppBar = () => {
    return (
        <div className="notes__appbar">
            <span>28 August 2021</span>
            <div>
                <button className='btn'>
                    Pictures
                </button>
            </div>
        </div>
    )
}
