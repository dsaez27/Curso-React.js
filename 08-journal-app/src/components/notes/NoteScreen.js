import React from 'react';
import { NotesAppBar } from './NotesAppBar';

export const NoteScreen = () => {
    return (
        <div className='notes__main-content'>
            <NotesAppBar />
            <div className='notes__content'>
                <input
                    className='notes__title-input'
                    type='text'
                    placeholder='Some awesome title'
                    autoComplete='off'
                />
                <textarea
                    placeholder='What happened today?'
                    className='notes__textarea'
                ></textarea>
                <div className='notes__image'>
                    <img src='https://i.blogs.es/2e7900/img19_1920x1200/1366_2000.jpg' alt='img'/>
                </div>
            </div>
        </div>
    );
};
