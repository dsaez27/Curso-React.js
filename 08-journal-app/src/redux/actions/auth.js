import Swal from 'sweetalert2';
import { types } from '../types';
import { finishLoading, setError, startLoading } from './ui';
import { noteLogout } from './notes';
import { firebase, googleAuthProvider } from '../../firebase/firebaseConfig';

export const startLoginEmailPassword = (email, password) => {
    return (dispatch) => {
        dispatch(startLoading());
        try {
            return firebase
                .auth()
                .signInWithEmailAndPassword(email, password)
                .then(({ user }) => {
                    dispatch(login(user.uid, user.displayName));
                    dispatch(finishLoading());
                })
                .catch((e) => {
                    dispatch(finishLoading());
                    Swal.fire({
                        title: 'Error',
                        text: e.message,
                        icon: 'error',
                    });
                });
        } catch (e) {
            console.log(e);
        }
    };
};

export const startRegisterEmailPasswordName = (email, password, name) => {
    return async (dispatch) => {
        try {
            await firebase
                .auth()
                .createUserWithEmailAndPassword(email, password)
                .then(async ({ user }) => {
                    await user.updateProfile({
                        displayName: name,
                    });

                    dispatch(login(user.uid, user.displayName));
                })
                .catch(({ message }) => {
                    Swal.fire({ title: 'Error', text: message, icon: 'error' });
                    dispatch(setError(message));
                });
        } catch (e) {
            console.log(e);
        }
    };
};

export const startGoogleLogin = () => {
    return async (dispatch) => {
       
            await firebase
                .auth()
                .signInWithPopup(googleAuthProvider)
                .then(({ user }) => {
                    dispatch(login(user.uid, user.displayName));
                })
                .catch((error) => {
                    console.log(error);
                });
         
    };
};

export const login = (uid, displayName) => {
    return {
        type: types.login,
        payload: {
            uid,
            displayName,
        },
    };
};

export const startLogout = () => {
    return (dispatch) => {
        firebase.auth().signOut();

        dispatch(logout());
        dispatch(noteLogout());
    };
};

export const logout = () => ({
    type: types.logout,
});
