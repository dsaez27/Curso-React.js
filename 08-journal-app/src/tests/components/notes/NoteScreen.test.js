import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import '../../../setupTest';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { NoteScreen } from './../../../components/notes/NoteScreen';
import { activeNote, startDeleting } from '../../../redux/actions/notes';

jest.mock('../../../redux/actions/notes', () => ({
    activeNote: jest.fn(),
    startDeleting: jest.fn(),
}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {
    auth: {
        uid: '123',
        name: 'test',
    },
    ui: {
        loading: false,
        msgError: null,
    },
    notes: {
        active: {
            id: 1,
            title: 'test',
            body: 'Mundo',
            date: 0,
        },
        notes: [],
    },
};

let store = mockStore(initState);
store.dispatch = jest.fn();

const wrapper = mount(
    <Provider store={store}>
        <NoteScreen />
    </Provider>
);

describe('Pruebas en el NoteScreen', () => {
    test('Debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de disparar el activaNote', () => {
        wrapper.find('input[name="title"]').simulate('change', {
            target: {
                name: 'title',
                value: 'test',
            },
        });
        expect(activeNote).toHaveBeenLastCalledWith(1, {
            body: 'Mundo',
            title: 'test',
            id: 1,
            date: 0,
        });
    });
});
