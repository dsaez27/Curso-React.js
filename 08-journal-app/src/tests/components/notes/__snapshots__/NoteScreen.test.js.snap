// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el NoteScreen Debe de mostrarse correctamente 1`] = `
<div
  className="notes__main-content"
>
  <div
    className="notes__appbar"
  >
    <span>
      28 August 2021
    </span>
    <input
      id="fileSelector"
      name="file"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div>
      <button
        className="btn"
        onClick={[Function]}
      >
        Pictures
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="notes__content"
  >
    <input
      autoComplete="off"
      className="notes__title-input"
      name="title"
      onChange={[Function]}
      placeholder="Some awesome title"
      type="text"
      value="test"
    />
    <textarea
      className="notes__textarea"
      name="body"
      onChange={[Function]}
      placeholder="What happened today?"
      value="Mundo"
    />
  </div>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
